ui/src/settings.vue

```diff
@@ -403,14 +507,134 @@ export default {
             this.passwordSuggestions = feedback.suggestions;
             this.passwordWarning = feedback.warning;
         },
+        loadUsers() {
+            if(!this.users.length) {
+                this.$http.get(Vue.config.auth_api+"/users").then(res=>{
+                    this.users = res.data;
+                }).catch(err=>{
+                    console.error(err.response);
+                    this.$notify({type: "error", text: err});
+                });
+            }
+            if(this.queryUser.length) return this.filteredUsers;
+            return this.users;
+        },
+        loadGroups() {
+            if(!this.groups.length) {
+                this.$http.get(Vue.config.auth_api+"/groups").then(res=>{
+                    this.groups = res.data;
+                    this.groups.forEach(group=>{
+                        if(group.admins) group.admins = group.admins.map(admins=>admins.sub);
+                        if(group.members) group.members = group.members.map(members=>members.sub);
+                    });
+                }).catch(err=>{
+                    console.error(err.response);
+                    this.$notify({type: "error", text: err});
+                });
+            }
+            if(this.queryGroup.length) return this.filteredGroups;
+            return this.groups;
+        },
+        selectUser(user) {
+            this.userEdit = Object.assign({}, this.userEdit, user);
+        },
+        selectGroup(group) {
+            this.groupEdit = Object.assign({}, this.groupEdit, group);
+        },
+        rowClass(item, type) {
+            if (!item || type !== 'row') return;
+            // if (item._id == this.userEdit._id || this.groupEdit._id == item._id) return 'table-success'
+            if(this.userEdit && this.userEdit._id == item._id) return 'table-primary';
+            if(this.groupEdit && this.groupEdit._id == item._id) return 'table-primary';
+        },
+        submitUser(e) {
+            e.preventDefault();
+            this.$http.put(Vue.config.auth_api+"/user/"+this.userEdit.sub,this.userEdit).then(res=>{
+                this.$notify({type: "success", text: res.data.message});
+                this.userEdit = null;
+            }).catch(console.error);
+        },
+        submitGroup(e) {
+            e.preventDefault();
+            if(!this.groupEdit.id) {
```

Comment:
I think I prefer using if/else instead of return here. You can also switch the blocks so you can simplify the if condition (remove the !)

```
            if(this.groupEdit.id) {
                this.$http.put(Vue.config.auth_api+"/group/"+this.groupEdit.id,this.groupEdit).then(res=>{
                    this.$notify({type: "success", text: res.data.message});
                }).catch(console.error);                
            } else {
                this.$http.post(Vue.config.auth_api+"/group/",this.groupEdit).then(res=>{
                    this.$notify({type: "success", text: res.data.message});
                }).catch(console.error);
            }
            this.groupEdit = null;
```